### PR TITLE
Freebayes 1.0.2  (new formula)

### PR DESCRIPTION
**Pull request by tseemann**
FreeBayes is a Bayesian genetic variant detector designed to find small polymorphisms, specifically SNPs (single-nucleotide polymorphisms), indels (insertions and deletions), MNPs (multi-nucleotide polymorphisms), and complex events (composite insertion and substitution events) smaller than the length of a short-read sequencing alignment.

### CODE REVIEW

**Comment by tseemann** [comment]
@sjackman The failed tests don't seem to be things to do with the formula/software itself? They seem to be check bugs?

**Comment by sjackman** [comment]
```
==> Downloading https://homebrew.bintray.com/bottles-science/vcflib-1.0.0.yosemite.bottle.tar.gz
Warning: Formula reports different SHA256: d9453aa105682ac17539efd11983b2d7018327cf58a6c19807613a8f0b2d959b
Downloaded to: /Library/Caches/Homebrew/vcflib-1.0.0.yosemite.bottle.tar.gz
SHA1: c8fb7b25435e9c0021c57f339184fff40947d2cf
SHA256: 33c51cf09ed90edfe7b15c7c8ae06a781d249a9ed14e0677a22aae3f3e401ae5
```

The SHA256 of the `vcflib` bottles for both Yosemite and Mavericks don't seem to be correct. I'm not sure how that happened. The quick fix is to change the `sha256` in `vcflib` to match the files stored on `bintray`. It may be worth a little sleuthing to try and figure out how that happened.

**Comment by sjackman** [comment]
To be on the safe side it might be an idea to rebuild all the `vcflib` bottles and bump the bottle revision.

**Comment by sjackman** [comment]
To answer your original question though, it doesn't appear to affect `freebayes`. Its bottles were built correctly. `vcflib` was compiled from source on the architectures with mismatching bottle `sha256`.

**Comment by tseemann** [comment]
@sjackman - can you confirm that this finally, actually built?  
IIRC you said the TRAVIS-CI is not critical?
My original attempts to get this through date back a year!

**Comment by sjackman** [comment]
Yes, it looks like it built to me.

Travis is not critical, but it worth clicking on the details to see why it failed.
https://travis-ci.org/Homebrew/homebrew-science/jobs/108952786#L1009
```
Error: GitHub API rate limit exceeded for 208.78.110.199. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
Try again in 24 minutes 2 seconds, or create a personal access token:
  https://github.com/settings/tokens/new?scopes=&description=Homebrew
```

**Comment by tseemann** [comment]
@BrewTestBot test this please

**Comment by yzhzu** [comment]
../vcflib/src/Variant.h:18:21: fatal error: tabix.hpp: No such file or directory
 #include "tabix.hpp"
                     ^
compilation terminated.
make[1]: *** [Makefile:107: freebayes.o] Error 1
make[1]: Leaving directory '/tmp/freebayes-20160814-25019-vljcga/src'
make: *** [Makefile:2: all] Error 2

who can help me solve the issue about freebayes? I installed it several times and did't know how to deal with the errors. thanks

File: freebayes.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  url "https://github.com/ekg/freebayes.git",
+    :tag => "v1.0.2",
+    :revision => "0cb269728b2db6307053cafe6f913a8b6fa1331e"
+
+  head "https://github.com/ekg/freebayes.git"
+
+  depends_on "cmake" => :build
+
+  # thes are needed for parallel freebayes
```
Comment (sjackman) [inline freebayes.rb]: typo

File: freebayes.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    inreplace "scripts/fasta_generate_regions.py", "#!/usr/bin/python", "#!/usr/bin/env python"
+
+    system "make"
+
+    bin.install "bin/freebayes"
+    bin.install "bin/bamleftalign"
+    bin.install "scripts/freebayes-parallel"
+    bin.install "scripts/coverage_to_regions.py"
+    bin.install "scripts/generate_freebayes_region_scripts.sh"
+    bin.install "scripts/fasta_generate_regions.py"
```
Comment (sjackman) [inline freebayes.rb]: Can you also install any included documentation with `doc.install`? (if it has some)

**Comment by sjackman** [comment]
Try
```sh
HOMEBREW_MAKE_JOBS=1 brew install freebayes
```

**Comment by sjackman** [comment]
Opened a new PR derived from this one at #4057

**Comment by tseemann** [comment]
#4057 fixed this mess. thank you!

**Pull request closed without merging**